### PR TITLE
Localdev part 2 

### LINKED ISSUE #1606: armadaproject.io: revise developer guide with instructions using a docker-compose approach

### PR DESCRIPTION
**Pull request by suprjinx**
Closes #1580 

Closes #1606

### CODE REVIEW

File: docs/developer.md [hunk 1]
```diff
@@ -15,14 +15,14 @@ To follow this section it is assumed you have:
...
+##### Kind + Docker Compose
+You can use a [kind](https://github.com/kubernetes-sigs/kind) Kubernetes cluster with Docker Compose to run a development stack.
  - The advantage of this is it is more like a real kubernetes cluster
- - However it is a bit more effort to set it up and it can only emulate a cluster with as much resource as your computer has
+ - However it can only emulate a small cluster
  
 This is recommended if load doesn't matter or you are working on features that rely on integrating with kubernetes functionality
   
-##### Fake-executor
+##### Fake-executorg
```
Comment (kannon92) [inline docs/developer.md:25]:
```suggestion
##### Fake-executor
```

File: e2e/setup/kind.yaml [hunk 2]
```diff
@@ -12,8 +14,11 @@ nodes:
         node-labels: "ingress-ready=true"
   extraPortMappings:
   - containerPort: 80 # port of the nginx ingress controller
-    hostPort: 5000 # expose ingresses on localhost:5000
+    hostPort: 5001 # expose ingresses on localhost:5000
```
Comment (suprjinx) [inline e2e/setup/kind.yaml:17]: MacOS uses 5000 for the system prefs

File: docs/developer.md [hunk 3]
```diff
@@ -31,39 +31,12 @@ You can use fake-executor
  
 This is recommended when working on features that are purely Armada specific or if you want to get a high load of jobs running through Armada components 
 
-#### Setup Kind development
+#### Setup Kind + Docker Compose development
 
-1. Get kind (Installation help [here](https://kind.sigs.k8s.io/docs/user/quick-start/))
+1. Start the "localdev" docker-compose environment
```
Comment (kannon92) [inline docs/developer.md:36]: Should we keep the kind installation instructions?  

Or does your script install it for you?

File: docs/developer.md [hunk 4]
```diff
@@ -89,69 +62,23 @@ This is recommended when working on features that are purely Armada specific or
...
-
-For Armada configuration check end to end test setup:
-```bash
-go run ./cmd/armada/main.go --config ./e2e/setup/insecure-armada-auth-config.yaml --config ./e2e/setup/nats/armada-config.yaml
-```
-
 ##### Lookout - Armada UI
-Lookout requires Armada to be configured with NATS Streaming.
-To run Lookout, firstly build frontend:
+Lookout requires Postgres to be up and running. Additionally, the react app must be built.
```
Comment (kannon92) [inline docs/developer.md:66]: Technically, Lookout requires both postgres and pulsar.  But I don't know if we really need this if someone runs the localdev script.  

Doesn't your script also build this react app?

File: e2e/setup/kind.yaml [hunk 5]
```diff
@@ -2,6 +2,8 @@ kind: Cluster
 apiVersion: kind.x-k8s.io/v1alpha4
 name: armada-test
 nodes:
+- role: worker
+  image: "kindest/node:v1.21.10"
```
Comment (kannon92) [inline e2e/setup/kind.yaml:6]:
```suggestion
  image: "kindest/node:v1.21.1"
```

File: docs/developer.md (hunk 4 above)
Comment (suprjinx) [reply docs/developer.md:66]: My script does not do the npm build -- you have to do the npm steps (right below this change). I guess I was being literal-minded about the dependencies (lookout-ingester needs pulsar, not lookout  - and NATS is no longer needed)

File: e2e/setup/kind.yaml (hunk 5 above)
Comment (suprjinx) [reply e2e/setup/kind.yaml:6]: ah thanks, moving both to .10

**Review by kannon92 — APPROVED** [review]
Once the integration tests pass, I say it is good to go!

**Pull request merged**